sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/CompressibleColumnBuilder.scala

```diff
@@ -61,8 +63,12 @@ private[columnar] trait CompressibleColumnBuilder[T <: AtomicType]
     super.initialize(initialSize, columnName, useCompression)
   }
 
+  // The various compression schemes, while saving memory use, cause all of the data within
+  // the row to become unaligned, thus causing crashes.  Until a way of fixing the compression
+  // is found to also allow aligned accesses this must be disabled for SPARC.
+
   protected def isWorthCompressing(encoder: Encoder[T]) = {
-    encoder.compressionRatio < 0.8
+    encoder.compressionRatio < 0.8 && !misalignedArch.contains(System.getProperty("os.arch"))
```

Comment:
You can cache whether the second clause check is true or false rather than checking it every time and looking up a system property. Probably even easier than storing a list of misaligned architectures. Can this be in an `object` anyway?